src/main/java/org/example/FlashCardArchive.java

```diff
@@ -0,0 +1,246 @@
+                menu();
+            }
+            case "ask": {
+                ask();
+                menu();
+            }
+            case "exit": {
+                System.out.println(writeLog(LogState.OUTPUT, "Bye bye!"));
+                if (!terminalArguments.isEmpty() && terminalArguments.contains("-export")) {
+                    for (int i = 0; i < terminalArguments.size(); i++) {
+                        if (terminalArguments.get(i).equals("-export")) {
+                            exportCards(terminalArguments.get(i + 1));
+                            break;
+                        }
+                    }
+                }
+                System.exit(0);
+            }
+            case "log": {
+                exportLog();
+                menu();
+            }
+            case "hardest card": {
+                hardestCard();
+                menu();
+            }
+            case "reset stats": {
+                resetErrors();
+                menu();
+            }
+        }
+    }
+
+
+    static void addCard() {
+        String term;
+        String def;
+        System.out.println(writeLog(LogState.OUTPUT, "The Card:"));
+        term = scanner.nextLine();
+        writeLog(LogState.INPUT, term);
+        if (flashCards.containsTerm(term)) {
+            System.out.println(writeLog(LogState.OUTPUT, "The card \"" + term + "\" already exists."));
+            return;
+        }
+        System.out.println(writeLog(LogState.OUTPUT, "The definition of the card:"));
+        def = scanner.nextLine();
+        writeLog(LogState.INPUT, def);
+        if (flashCards.containsDefinition(def)) {
+            System.out.println(writeLog(LogState.OUTPUT, "The definition \"" + def + "\" already exists."));
+            return;
+        }
+        flashCards.addFlashCard(new FlashCard(term, def));
+        System.out.println(writeLog(LogState.OUTPUT, "The pair (\"" + term + "\":\"" + def + "\") has been added"));
+    }
+
+    static void removeCard() {
+        System.out.println(writeLog(LogState.OUTPUT, "Which Card?"));
+        String term = scanner.nextLine();
+        writeLog(LogState.INPUT, term);
+        if (!flashCards.containsTerm(term)) {
+            System.out.println(writeLog(LogState.OUTPUT, "Can't remove \"" + term + "\": there is no such card."));
+        } else {
+            flashCards.getInstance().remove(flashCards.findByTerm(term));
+            System.out.println(writeLog(LogState.OUTPUT, "The card has been removed."));
+        }
+    }
+
+    static void importCards(String filePath) throws IOException {
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filePath));
+        } catch (FileNotFoundException e) {
+            System.out.println(writeLog(LogState.OUTPUT, "\"File not found.\""));
+            menu();
+        }
+        int count = 0;
+        while (true) {
+            assert sc != null;
+            if (!sc.hasNextLine()) break;
```

Comment:
in general methods are slightly unreadable. I mean here for example it would be great to have couple of blank lines between some ifs.